OsmAnd/src/net/osmand/plus/activities/MapActivityActions.java

```diff
@@ -630,50 +638,77 @@ public void onPrepareDialog(int id, Dialog dialog) {
 		}
 	}
 
-
 	public ContextMenuAdapter createMainOptionsMenu() {
 		final OsmandMapTileView mapView = mapActivity.getMapView();
 		final OsmandApplication app = mapActivity.getMyApplication();
 		ContextMenuAdapter optionsMenuHelper = new ContextMenuAdapter();
 		boolean nightMode = getMyApplication().getDaynightHelper().isNightModeForMapControls();
+		
+		if (drawerMode == DRAWER_MODE_SWITCH_PROFILE) {
+			return createSwitchProfileOptionsMenu(app, optionsMenuHelper, nightMode);
+		}
+		return createNormalOptionsMenu(app, optionsMenuHelper, nightMode);
+	}
 
-		//switch profile button
+	private ContextMenuAdapter createSwitchProfileOptionsMenu(final OsmandApplication app, ContextMenuAdapter optionsMenuHelper, boolean nightMode) {
+		
+		createProfilesController(app, optionsMenuHelper, nightMode, true);
+		
+		List<ApplicationMode> activeModes = ApplicationMode.values(app);
 		ApplicationMode currentMode = app.getSettings().APPLICATION_MODE.get();
+
 		String modeDescription;
-		if (currentMode.isCustomProfile()) {
-			modeDescription = String.format(app.getString(R.string.profile_type_descr_string),
-					Algorithms.capitalizeFirstLetterAndLowercase(currentMode.getParent().toHumanString(app)));
-		} else {
-			modeDescription = getString(R.string.profile_type_base_string);
+		
+		for (final ApplicationMode appMode : activeModes) {
+			if (appMode.isCustomProfile()) {
+				modeDescription = String.format(app.getString(R.string.profile_type_descr_string),
+						Algorithms.capitalizeFirstLetterAndLowercase(appMode.getParent().toHumanString(app)));
+			} else {
+				modeDescription = getString(R.string.profile_type_base_string);
+			}
+
+			int tag = currentMode.equals(appMode) ? PROFILES_CHOSEN_PROFILE_TAG : PROFILES_NORMAL_PROFILE_TAG;
+
+			optionsMenuHelper.addItem(new ItemBuilder().setLayout(R.layout.profile_list_item)
+					.setIcon(appMode.getIconRes())
+					.setColor(appMode.getIconColorInfo().getColor(nightMode))
+					.setTag(tag)
+					.setTitle(appMode.toHumanString(app))
+					.setDescription(modeDescription)
+					.setListener(new ItemClickListener() {
+						@Override
+						public boolean onContextMenuClick(ArrayAdapter<ContextMenuItem> adapter, int itemId, int position, boolean isChecked, int[] viewCoordinates) {
+							app.getSettings().APPLICATION_MODE.set(appMode);
+							drawerMode = DRAWER_MODE_NORMAL;
```

Comment:
Is drawerMode become normal after pressing button only? What if we close and open the drawer. Should it be reset in this case also?